metricbeat/beater/config.go

```diff
@@ -1,9 +1,29 @@
 package beater
 
-import "github.com/elastic/beats/libbeat/common"
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+)
 
 // Config is the root of the Metricbeat configuration hierarchy.
 type Config struct {
 	// Modules is a list of module specific configuration data.
-	Modules []*common.Config `config:"modules" validate:"required"`
+	Modules       []*common.Config    `config:"modules"`
+	ReloadModules ModulesReloadConfig `config:"reload.modules"`
```

Comment:
You'll need to make `ReloadModules` a pointer too. But with the default present it won't be nil.